src/main/java/nextstep/qna/domain/Question.java

```diff
@@ -1,92 +1,71 @@
 package nextstep.qna.domain;
 
+import nextstep.qna.CannotDeleteException;
+import nextstep.qna.domain.base.BaseDate;
 import nextstep.users.domain.NsUser;
 
-import java.time.LocalDateTime;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 
-public class Question {
-    private Long id;
-
-    private String title;
-
-    private String contents;
+import static java.time.LocalDateTime.now;
+import static nextstep.qna.domain.ContentType.QUESTION;
 
-    private NsUser writer;
+public class Question extends BaseDate {
 
-    private List<Answer> answers = new ArrayList<>();
-
-    private boolean deleted = false;
+    private Long id;
 
-    private LocalDateTime createdDate = LocalDateTime.now();
+    private Content content;
 
-    private LocalDateTime updatedDate;
+    private Answers answers = new Answers();
 
     public Question() {
     }
 
-    public Question(NsUser writer, String title, String contents) {
-        this(0L, writer, title, contents);
+    public Question(Long id, Content content) {
+        this.id = id;
+        this.content = content;
     }
 
-    public Question(Long id, NsUser writer, String title, String contents) {
-        this.id = id;
-        this.writer = writer;
-        this.title = title;
-        this.contents = contents;
+    public static Question toQuestion(NsUser writer, String title, String contents) {
+        return new Question(0L, new Content(title, contents, writer));
     }
 
     public Long getId() {
         return id;
     }
 
-    public String getTitle() {
-        return title;
-    }
-
-    public Question setTitle(String title) {
-        this.title = title;
-        return this;
-    }
-
-    public String getContents() {
-        return contents;
-    }
-
-    public Question setContents(String contents) {
-        this.contents = contents;
-        return this;
-    }
-
-    public NsUser getWriter() {
-        return writer;
-    }
-
     public void addAnswer(Answer answer) {
         answer.toQuestion(this);
-        answers.add(answer);
+        answers.addAnswer(answer);
     }
 
-    public boolean isOwner(NsUser loginUser) {
-        return writer.equals(loginUser);
+    public List<DeleteHistory> delete(NsUser loginUser) throws CannotDeleteException {
+        if (!content.isOwner(loginUser)) {
+            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
+        }
+        content.delete();
```

Comment:
Content가 작성자, 삭제여부 모두 가지고 있으니 메시지를 보내면 어떨까요?
```suggestion
        content.delete(loginUser);
```